ttnn/cpp/ttnn/operations/experimental/ssm/prefix_scan/device/prefix_scan.hpp

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/run_operation.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::experimental::ssm {
+
+struct PrefixScan {
+    MemoryConfig memory_config;
+    DataType dtype;
+    MathFidelity math_fidelity;
+
+    void validate(const std::vector<Tensor>& input_tensors) const;
+    std::vector<tt::tt_metal::Shape> compute_output_shapes(const std::vector<Tensor>& input_tensors) const;
+    std::vector<Tensor> create_output_tensors(const std::vector<Tensor>& input_tensors) const;
+    operation::ProgramWithCallbacks create_program(
+        const std::vector<Tensor>& input_tensors, std::vector<Tensor>& output_tensors) const;
+};
+
+inline Tensor prefix_scan(
```

Comment:
this should be removed from here and be inside operator() in ttnn/cpp/ttnn/operations/experimental/ssm/prefix_scan/prefix_scan.hpp/cpp

Reply:
Or should be removed completely, it does not seem to be used and should not be used in composite.